httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/FormBodyPartBuilder.java

```diff
@@ -96,12 +99,61 @@ public FormBodyPartBuilder setField(final String name, final String value) {
         return this;
     }
 
+    /**
+     * Sets the multipart mode for this builder, determining how filenames are encoded in the
+     * {@code Content-Disposition} header. The mode affects whether the {@code filename*} parameter
+     * is included for non-ISO-8859-1 filenames.
+     * <p>
+     * In {@link HttpMultipartMode#LEGACY} mode, only the {@code filename} parameter is included
+     * with the raw value, mimicking pre-RFC 7578 behavior. In {@link HttpMultipartMode#STRICT}
+     * or {@link HttpMultipartMode#EXTENDED} modes, the {@code filename*} parameter is added
+     * with RFC 5987 encoding for filenames containing non-ISO-8859-1 characters.
+     * </p>
+     * <p>
+     * If {@code mode} is {@code null}, it defaults to {@link HttpMultipartMode#STRICT}.
+     * </p>
+     *
+     * @param mode the {@link HttpMultipartMode} to use, or {@code null} for default behavior
+     * @return this builder instance for method chaining
+     * @since 5.5
+     */
+    public FormBodyPartBuilder setMode(final HttpMultipartMode mode) {
```

Comment:
@arturobernalg Would not it be better to pass this parameter at the construction time to `#create` method?